src/components/FilterList/SearchSettingsMenu/FilterListSearchSettingsMenu.tsx

```diff
@@ -0,0 +1,35 @@
+import { ActionIcon, Checkbox, Group, Menu, Stack } from '@mantine/core';
+
+import { SettingsIcon } from '@/icons/icons';
+import { camelCaseToRegularText } from '@/util/text';
+
+interface Props<T> {
+  keys: Record<keyof T, boolean>;
+  setKey: (key: keyof T, enabled: boolean) => void;
+}
+
+export function FilterListSearchSettingsMenu<T>({ keys, setKey }: Props<T>) {
+  return (
+    <Menu position={'right-start'} withArrow closeOnItemClick={false}>
+      <Menu.Target>
+        <ActionIcon flex={'0 0 auto'}>
+          <SettingsIcon />
+        </ActionIcon>
+      </Menu.Target>
+      <Menu.Dropdown maw={'300px'}>
+        <Menu.Label>Search these keys</Menu.Label>
+        <Stack p={'xs'}>
+          {Object.entries(keys).map(([key, enabled]) => (
```

Comment:
Now it's possible to uncheck all the checkboxes => there will be no result in the search and it's hard to know why. 

Should we maybe 
1. Prevent unchecking all checkboxes? 
2. or, at least show some type of warning if no checkbox is checked? The settings icon could change to a warning, for example, and some warning text could be shown in the menu